src/test/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/SampleServiceImplTest.java

```diff
@@ -0,0 +1,113 @@
+import uk.gov.ons.ctp.response.collection.exercise.domain.ExerciseSampleUnitGroup;
+import uk.gov.ons.ctp.response.collection.exercise.repository.CollectionExerciseRepository;
+import uk.gov.ons.ctp.response.collection.exercise.repository.SampleUnitGroupRepository;
+import uk.gov.ons.ctp.response.collection.exercise.repository.SampleUnitRepository;
+import uk.gov.ons.ctp.response.collection.exercise.representation.CollectionExerciseDTO.CollectionExerciseEvent;
+import uk.gov.ons.ctp.response.collection.exercise.representation.CollectionExerciseDTO.CollectionExerciseState;
+import uk.gov.ons.ctp.response.collection.exercise.representation.SampleUnitGroupDTO.SampleUnitGroupState;
+import uk.gov.ons.ctp.response.sample.representation.SampleUnitDTO.SampleUnitType;
+import uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SampleServiceImplTest {
+  private static final UUID COLLEX_ID = UUID.randomUUID();
+  private static final UUID SAMPLE_ID = UUID.randomUUID();
+
+  @Mock private SampleUnitRepository sampleUnitRepo;
+
+  @Mock private SampleUnitGroupRepository sampleUnitGroupRepo;
+
+  @Mock private CollectionExerciseRepository collectRepo;
+
+  @Mock
+  private StateTransitionManager<CollectionExerciseState, CollectionExerciseEvent>
+      collectionExerciseTransitionState;
+
+  @InjectMocks private SampleServiceImpl underTest;
+
+  @Test
+  public void testAcceptSampleUnit_CountNotEqual() throws CTPException {
+    CollectionExercise collex = new CollectionExercise();
+    collex.setSampleSize(50);
+    collex.setState(CollectionExerciseState.EXECUTION_STARTED);
+
+    acceptSampleUnitWithCollex(collex);
+
+    verify(collectRepo, never()).saveAndFlush(any());
+    verify(collectionExerciseTransitionState, never()).transition(any(), any());
+  }
+
+  @Test
+  public void testAcceptSampleUnit_CountEqual() throws CTPException {
+    CollectionExercise collex = new CollectionExercise();
+    collex.setSampleSize(99);
+    collex.setState(CollectionExerciseState.EXECUTION_STARTED);
+
+    when(collectionExerciseTransitionState.transition(any(), any()))
+        .thenReturn(CollectionExerciseState.EXECUTED);
+
+    acceptSampleUnitWithCollex(collex);
+
+    ArgumentCaptor<CollectionExercise> collexArgumentCaptor =
+        ArgumentCaptor.forClass(CollectionExercise.class);
+    verify(collectRepo).saveAndFlush(collexArgumentCaptor.capture());
+    assertEquals(CollectionExerciseState.EXECUTED, collexArgumentCaptor.getValue().getState());
+    assertNotNull(collexArgumentCaptor.getValue().getActualExecutionDateTime());
+  }
+
+  private void acceptSampleUnitWithCollex(CollectionExercise collex) throws CTPException {
+    SampleUnit sampleUnit =
+        SampleUnit.builder()
+            .withId(SAMPLE_ID.toString())
+            .withFormType("X")
+            .withSampleUnitRef("REF123")
+            .withSampleUnitType("B")
+            .withCollectionExerciseId(COLLEX_ID.toString())
```

Comment:
This strikes me as odd as a CollectionExercise gets passed in as a parameter - I'd expect collex.getId().toString().  I can see it works but it doesn't look right.

Reply:
have changed it